astropy/convolution/boundary_none.pyx

```diff
@@ -59,17 +59,29 @@ def convolve1d_boundary_none(np.ndarray[DTYPE_t, ndim=1] f,
     # GIL acquired again here
     return conv
 
-
-@cython.boundscheck(False)  # turn off bounds-checking for entire function
+@cython.boundscheck(False)
 def convolve2d_boundary_none(np.ndarray[DTYPE_t, ndim=2] f,
                              np.ndarray[DTYPE_t, ndim=2] g,
-                             bint normalize_by_kernel):
+                             bint nan_interpolate):
 
     if g.shape[0] % 2 != 1 or g.shape[1] % 2 != 1:
         raise ValueError("Convolution kernel must have odd dimensions")
 
     assert f.dtype == DTYPE and g.dtype == DTYPE
 
+    if nan_interpolate:
+        return convolve2d_boundary_none_internal(f, g, True)
+    else:
+        return convolve2d_boundary_none_internal(f, g, False)
+
+# NOTE: check inline is compulsory for complier opt, i.e. __attribute__((always_inline)) 
+@cython.cdivision(True)
+@cython.wraparound(False)
+@cython.boundscheck(False)  # turn off bounds-checking for entire function
+cdef inline convolve2d_boundary_none_internal(np.ndarray[DTYPE_t, ndim=2] f,
```

Comment:
Make inline so that the above wrapper can then expand it with the True/False literals. The C compiler will then optimize these conditionals out of the code.

Reply:
This allows the code to contained in a single place, not duplicated or manually expanded, whilst still creating performant code.